build/preprocess_template.py

```diff
@@ -34,25 +40,53 @@ def load_common_templates(path):
   except yaml.YAMLError as exc:
     raise exc 
 
-def process_annotations(commonTemplates, rules):
+def process_rules(commonTemplates, rules):
   for template in commonTemplates:
       metadata = template.get("metadata", {})
       annotations = metadata.get("annotations",{})
       labels = metadata.get("labels", {})
       annotationsAdded = []
-      for rule in rules:
-        key = rule.get("matchLabel")[0]
-        value = rule.get("matchLabel")[1]
-        if labels.get(key) == value:
-          annotationKey = rule.get("addAnnotation")[0]
-          annotationValue = rule.get("addAnnotation")[1]
-          annotations[annotationKey] = annotationValue
-          annotationsAdded.append(annotationKey + ": " +  annotationValue)
-
-      if len(annotationsAdded) > 0:
+      addAnnotationRules = rules.get("addAnnotation")
+      if addAnnotationRules != None:
+        for rule in addAnnotationRules:
+          key = rule.get("matchLabel")[0]
+          value = rule.get("matchLabel")[1]
+          if labels.get(key) == value:
+            annotationKey = rule.get("addAnnotation")[0]
+            annotationValue = rule.get("addAnnotation")[1]
+            annotations[annotationKey] = annotationValue
+            annotationsAdded.append(annotationKey + ": " +  annotationValue)
```

Comment:
do we use "annotationsAdded" just for logging? If so:

  annotationsAdded.append("adding " + annotationKey + ": " + annotationValue)